### PR TITLE
Support for the FD gauge

### PR DESCRIPTION
**Pull request by oliviermattelaer**
- adding a new gauge "set gauge FD"
- For this the model need to support the Feynman gauge and if it does the FD is constructed on the flight (by merging interaction/...)

### CODE REVIEW

**Comment by oliviermattelaer** [comment]
Hi Marco,

Any objection on this? I would like to move forward on 3.6.0.

Cheers,

Olivier

File: aloha/aloha_writers.py [hunk 1]
```diff
@@ -536,7 +543,11 @@ def get_declaration_txt(self):
         if KERNEL.has_pi:
             out.write(' parameter (PI=%s)\n' % self.change_number_format(cmath.pi))
         
-        
+        if aloha.unitary_gauge == 3: # Feynman
```
Comment (marcozaro) [inline aloha/aloha_writers.py]: Feynman->FD gauge?

File: bin/create_release.py [hunk 2]
```diff
@@ -172,7 +172,14 @@
     auto_update = False
 
 # checking that the rev_nb is in a reasonable range compare to the old one.
```
Comment (marcozaro) [inline bin/create_release.py]: there are some conflicts

File: madgraph/core/color_algebra.py [hunk 3]
```diff
@@ -957,6 +957,8 @@ def to_canonical(self, immutable=None):
     def __eq__(self, col_str):
         """Check if two color strings are equivalent by checking if their
         canonical representations and the coefficients are equal."""
+        #misc.sprint(self.coeff, self.Nc_power, self.is_imaginary, self.to_canonical())
```
Comment (marcozaro) [inline madgraph/core/color_algebra.py:960]: remove these two lines?

File: madgraph/interface/amcatnlo_run_interface.py [hunk 4]
```diff
@@ -5272,6 +5272,74 @@ def copy_lep_densities(self, name, sourcedir):
             files.cp(os.path.join(lep_d_path, filename), pdf_path)
 
 
+    def link_and_copy_epdf(self, pdlabel, lhaid, libdir):
```
Comment (marcozaro) [inline madgraph/interface/amcatnlo_run_interface.py]: why this change here? unrelated to the FD gauge, right?

**Review by marcozaro — CHANGES_REQUESTED** [review]
generally ok! nice work, see a bunch of minor comments

File: aloha/aloha_writers.py (hunk 1 above)
Comment (oliviermattelaer) [reply aloha/aloha_writers.py]: thanks

File: bin/create_release.py (hunk 2 above)
Comment (oliviermattelaer) [reply bin/create_release.py]: -> fix

File: madgraph/core/color_algebra.py (hunk 3 above)
Comment (oliviermattelaer) [reply madgraph/core/color_algebra.py:960]: That one I would prefer to keep since they are helpfull for debugging

File: madgraph/interface/amcatnlo_run_interface.py (hunk 4 above)
Comment (oliviermattelaer) [reply madgraph/interface/amcatnlo_run_interface.py]: Looks like this is a git error (or when solving a conflict), since now this function is define twice within this file (line 5207 and 5275)... Thanks for spotting this.

**Review by oliviermattelaer — COMMENTED** [review]
Thanks Marco,

I have implement all the change that you spot (thanks so much for detecting that I mess a up on one of merging of this branch). I have just not act on your request to remove a comment since I find it usefull for debugging if/when needed.

Thanks,

Olivier

**Review by marcozaro — APPROVED** [review]

**Comment by marcozaro** [comment]
great!

Marco
